client/blocks/stats-navigation/index.js

```diff
@@ -61,6 +62,103 @@ function withNoticeHook( HookedComponent ) {
 	};
 }
 
+const SelectNav = ( {
+	label,
+	validNavItems,
+	interval,
+	slugPath,
+	adminUrl,
+	selectedItem,
+	showLock,
+	isLegacy,
+	showIntervals,
+	pathTemplate,
+} ) => {
+	return (
+		<>
+			<SectionNav selectedText={ label }>
+				<NavTabs selectedText={ label }>
+					{ validNavItems.map( ( item ) => {
+						const navItem = navItems[ item ];
+						const intervalPath = navItem.showIntervals ? `/${ interval || 'day' }` : '';
+						const itemPath = `${ navItem.path }${ intervalPath }${ slugPath }`;
+						const className = 'stats-navigation__' + item;
+						if ( item === 'store' && config.isEnabled( 'is_running_in_jetpack_site' ) ) {
+							return (
+								<NavItem
+									className={ className }
+									key={ item }
+									onClick={ () =>
+										( window.location.href = `${ adminUrl }admin.php?page=wc-admin&path=%2Fanalytics%2Foverview` )
+									}
+									selected={ false }
+								>
+									{ navItem.label }
+								</NavItem>
+							);
+						}
+						return (
+							<NavItem
+								className={ className }
+								key={ item }
+								path={ itemPath }
+								selected={ selectedItem === item }
+							>
+								{ navItem.label }
+								{ navItem.paywall && showLock && ' 🔒' }
+							</NavItem>
+						);
+					} ) }
+				</NavTabs>
+
+				{ isLegacy && showIntervals && (
+					<Intervals selected={ interval } pathTemplate={ pathTemplate } />
+				) }
+			</SectionNav>
+
+			{ isLegacy && showIntervals && (
+				<Intervals selected={ interval } pathTemplate={ pathTemplate } standalone />
+			) }
+		</>
+	);
+};
+
+const TabNav = ( { validNavItems, interval, slugPath, adminUrl, selectedItem, showLock } ) => {
+	const tabs = validNavItems.map( ( item ) => {
+		const navItem = navItems[ item ];
+		const intervalPath = navItem.showIntervals ? `/${ interval || 'day' }` : '';
+		const itemPath = `${ navItem.path }${ intervalPath }${ slugPath }`;
+		return {
+			name: item,
+			title: navItem.label + ( navItem.paywall && showLock ? ' 🔒' : '' ),
+			className: 'stats-navigation__' + item,
+			path: itemPath,
+		};
+	} );
+
+	return (
+		<TabPanel
+			className="stats-navigation__tabs"
+			tabs={ tabs }
+			onSelect={ ( tabName ) => {
+				const tab = tabs.find( ( { name } ) => name === tabName );
+
+				if ( tab.name === 'store' && config.isEnabled( 'is_running_in_jetpack_site' ) ) {
+					window.location.href = `${ adminUrl }admin.php?page=wc-admin&path=%2Fanalytics%2Foverview`;
+				} else if ( tab.path ) {
+					page( tab.path );
+				}
+			} }
+			initialTabName={ selectedItem }
+		>
+			{ () => (
+				// Placeholder div since content is rendered elsewhere
+				<div className="stats-navigation__content" />
```

Comment:
No need to render anything really, and it could use an empty function like `()=>null`